### PR TITLE
Add -haproxy-protocol startup parameter

### LINKED ISSUE #1024: Add -haproxy-protocol startup parameter
I know that there is a startup parameter of -port, but in some cases, multiple velocities may be synchronized with git, some of which will enable haproxy-protocol, and some will not.

This is a small update, thanks.

### PR DESCRIPTION
**Pull request by BingYanchi**
resolves https://github.com/PaperMC/Velocity/issues/1024

### CODE REVIEW

File: proxy/src/main/java/com/velocitypowered/proxy/ProxyOptions.java [hunk 1]
```diff
@@ -34,6 +34,7 @@ public final class ProxyOptions {
   private static final Logger logger = LogManager.getLogger(ProxyOptions.class);
   private final boolean help;
   private final @Nullable Integer port;
+  private final Boolean haproxy;
```
Comment (powercasgamer) [inline proxy/src/main/java/com/velocitypowered/proxy/ProxyOptions.java]:
```suggestion
  private final boolean haproxy;
```

File: proxy/src/main/java/com/velocitypowered/proxy/ProxyOptions.java [hunk 2]
```diff
@@ -64,4 +71,8 @@ boolean isHelp() {
   public @Nullable Integer getPort() {
     return this.port;
   }
+
+  public Boolean isHaproxy() {
```
Comment (powercasgamer) [inline proxy/src/main/java/com/velocitypowered/proxy/ProxyOptions.java]:
```suggestion
  public boolean isHaproxy() {
```

**Review by powercasgamer — CHANGES_REQUESTED** [review]

File: proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java [hunk 3]
```diff
@@ -243,6 +243,12 @@ void start() {
       this.cm.bind(configuration.getBind());
     }
 
+    final Boolean haproxy = this.options.isHaproxy();
+    if (haproxy != null) {
```
Comment (electronicboy) [inline proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java:261]: This is always true, as you declared the method return to be a primative

**Review by 4drian3d — APPROVED** [review]

**Comment by BingYanchi** [comment]
```
Exception in thread "main" java.lang.NullPointerException: Cannot read the array length because "<local2>" is null
        at joptsimple.ArgumentAcceptingOptionSpec.defaultsTo(ArgumentAcceptingOptionSpec.java:225)
        at com.velocitypowered.proxy.ProxyOptions.<init>(ProxyOptions.java:51)
        at com.velocitypowered.proxy.Velocity.main(Velocity.java:62)
```
There is a problem with this PR and it needs to be rolled back.

**Pull request merged**